### PR TITLE
Enable macOS arm64, using https://github.com/s-u/macosvm

### PR DESCRIPTION
**Pull request by BillyONeal**
Depends on https://github.com/microsoft/vcpkg/pull/35617

MacOS 13.3.

### CODE REVIEW

File: scripts/azure-pipelines/osx/README.md [hunk 1]
```diff
@@ -88,47 +90,59 @@ This is the checklist for what the vcpkg team does when updating the macOS machi
...
-    prlctl create "vcpkg-osx-<date>-arm64" -o macos --restore-image ~/Downloads/path/to.ipsw
-    prlctl set "vcpkg-osx-<date>-arm64" --memsize 12288 --cpus 8 --isolate-vm on --shf-host-defined off
-    cd ~/Parallels/vcpkg-osx-<date>-arm64.macvm
-    truncate -s 500G disk0.img
+    mkdir -p ~/Parallels/vcpkg-osx-2024-01-22-arm64
+    cd ~/Parallels/vcpkg-osx-2024-01-22-arm64
+    ~/macosvm --disk disk.img,size=500g --aux aux.img -c 8 -r 12g --restore ~/UniversalMac_14.3_23D56_Restore.ipsw ./vm.json
+    ~/macosvm -g ./vm.json
     ```
 - [ ] Start the VM in parallels and follow prompts as you would on real hardware.
```
Comment (autoantwort) [inline scripts/azure-pipelines/osx/README.md:109]: Is there a benefit of using parallels instead of `~/macosvm -g ./vm.json`?

Comment (BillyONeal) [reply scripts/azure-pipelines/osx/README.md:109]: Bill forgot to edit that line :)

**Comment by BillyONeal** [comment]
https://github.com/microsoft/vcpkg/pull/35617 wasn't clean so I merged this back into it.

File: scripts/azure-pipelines/osx/README.md [hunk 2]
```diff
@@ -187,21 +201,37 @@ Run these steps on each machine to add to the fleet. Skip steps that were done i
...
     scp ./register-guest.sh vcpkg@MACHINE:/Users/vcpkg
-    ssh vcpkg@MACHINE chmod +x /Users/vcpkg/register-guest.sh
-    ssh vcpkg@MACHINE /Users/vcpkg/register-guest.sh <PAT>
+    ssh vcpkg@MACHINE
+    chmod +x register-guest.sh
+    ./register-guest.sh <PAT>
+    rm register-guest.sh
+    exit
+    ```
+- [ ] In the VM's terminal, relaunch the VM in ephemeral mode with:
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply scripts/azure-pipelines/osx/README.md:232]: It should say KVM, it's talking about the host.

**Pull request closed without merging**